shared/src/main/scala/coop/rchain/shared/Resources.scala

```diff
@@ -11,10 +12,30 @@ object Resources {
     * @param a A given resource implementing [[AutoCloseable]]
     * @param f A function that takes this resource as its argument
     */
-  def withResource[A <: AutoCloseable, B](a: A)(f: A => B): B =
+  def withResource[A <: AutoCloseable, B](r: => A)(f: A => B): B = {
```

Comment:
Would be great to have unit tests for that. Would you mind adding them in this PR? If you'd prefer not, please file a ticket for adding them.

Reply:
Sorry, sure. Will add one.